tests/components/bluetooth_le_tracker/test_device_tracker.py

```diff
@@ -65,6 +66,152 @@ async def read_gatt_char(self, *args, **kwargs):
+            )
+            await hass.async_block_till_done()
+
+        # Advance time to trigger updates
+        time_after_consider_home = dt_util.utcnow() + config[CONF_CONSIDER_HOME] / 2
+        with patch(
+            "homeassistant.components.device_tracker.legacy.dt_util.utcnow",
+            return_value=time_after_consider_home,
+        ):
+            async_fire_time_changed(hass, time_after_consider_home)
+            await hass.async_block_till_done()
+
+        # Advance time over the consider home threshold and trigger update after the threshold
+        time_after_consider_home = dt_util.utcnow() + config[CONF_CONSIDER_HOME]
+        with patch(
+            "homeassistant.components.device_tracker.legacy.dt_util.utcnow",
+            return_value=time_after_consider_home,
+        ):
+            async_fire_time_changed(hass, time_after_consider_home)
+            await hass.async_block_till_done()
+
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == "not_home"
+
+
+async def test_see_device_if_time_updated(
+    hass: HomeAssistant,
+    mock_bluetooth: None,
+    mock_device_tracker_conf: list[legacy.Device],
+) -> None:
+    """Test device remaining home after consider_home threshold from first scan if the subsequent scans have incremented last seen time."""
+
+    address = "DE:AD:BE:EF:13:37"
+    name = "Mock device name"
+    entity_id = f"{DOMAIN}.{slugify(name)}"
+
+    with patch(
+        "homeassistant.components.bluetooth.async_discovered_service_info"
+    ) as mock_async_discovered_service_info, patch.object(
+        device_tracker, "MIN_SEEN_NEW", 3
+    ):
+        device = BluetoothServiceInfoBleak(
+            name=name,
+            address=address,
+            rssi=-19,
+            manufacturer_data={},
+            service_data={},
+            service_uuids=[],
+            source="local",
+            device=BLEDevice(address, None),
+            advertisement=generate_advertisement_data(local_name="empty"),
+            time=0,
+            connectable=False,
+        )
+        # Return with name with time = 0 initially
+        mock_async_discovered_service_info.return_value = [device]
+
+        config = {
+            CONF_PLATFORM: "bluetooth_le_tracker",
+            CONF_SCAN_INTERVAL: timedelta(minutes=1),
+            CONF_TRACK_NEW: True,
+            CONF_CONSIDER_HOME: timedelta(minutes=10),
+        }
+        result = await async_setup_component(hass, DOMAIN, {DOMAIN: config})
+        assert result
+
+        # Tick until device seen enough times for to be registered for tracking
+        for _ in range(device_tracker.MIN_SEEN_NEW):
+            async_fire_time_changed(
+                hass,
+                dt_util.utcnow() + config[CONF_SCAN_INTERVAL] + timedelta(seconds=1),
+            )
+            await hass.async_block_till_done()
+
+        # Increment device time so it gets seen in the next update
+        device.time = 1
```

Comment:
This will likely break in the future if `BluetoothServiceInfoBleak` becomes immutable or frozen